### PR TITLE
JWWEB-403: main action goes to page specific to current wallet

### PR DESCRIPTION
**Pull request by stcherenkov**
Sorry for bad commit names :(

The second one is actually MainAction fix, dunno where out git took this message

### CODE REVIEW

File: src/layouts/MenuLayout/MenuLayout.js [hunk 1]
```diff
@@ -7,9 +7,8 @@ import MenuPanel from 'components/MenuPanel'
 import OverlayNotification from 'components/OverlayNotification'
 
 type Props = {|
-  +openLayout: () => void,
-  +closeLayout: () => void,
-  +onSendAssetClick: () => void,
+  +openLayout: Function,
+  +closeLayout: Function,
```
Comment (IViol) [inline src/layouts/MenuLayout/MenuLayout.js]: why did you remove types for functions' signature?

File: src/components/MenuPanel/MainAction/MainAction.js [hunk 2]
```diff
@@ -4,25 +4,60 @@ import React from 'react'
...
+          onClick={null}
+          label='Add private key'
+          labelColor='dark'
+          color='white'
+        />
+      </Link>
+    )
+  }
+
+  if (isReadOnly && !isMnemonic) {
```
[the remark this answers is not in the document]
Comment (stcherenkov) [reply src/components/MenuPanel/MainAction/MainAction.js]: There are 3 cases, not 2.

Separating those cases is more explicit than nesting ternaries.

File: src/layouts/MenuLayout/MenuLayout.js (hunk 1 above)
Comment (stcherenkov) [reply src/layouts/MenuLayout/MenuLayout.js]: Because any signature would do the trick

File: src/components/MenuPanel/MainAction/MainAction.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (IViol) [reply src/components/MenuPanel/MainAction/MainAction.js]: Imagine you have component with 10 required props.

Separating render of component for each combination of these props is more explicit, than compute some attributes to render component once. Don't you agree?

And I'm sure that there are just 2 cases:
- Send asset
- Add access for readOnly wallet

File: src/layouts/MenuLayout/MenuLayout.js (hunk 1 above)
Comment (IViol) [reply src/layouts/MenuLayout/MenuLayout.js]: sorry, don't understand.
You just removed valid types for function signature.

`() => void` means that function doesn't get any params and return nothing. And it is true for specific redux actions

File: src/components/MenuPanel/MainAction/MainAction.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (stcherenkov) [reply src/components/MenuPanel/MainAction/MainAction.js]: 10 required props in simple UI component is most likely poor design with very high complexity.

If it is composite component encapsulating view logic, those props should be calculated in controller (or whatever works as a controller).

[the remark this answers is not in the document]
Comment (stcherenkov) [reply src/components/MenuPanel/MainAction/MainAction.js]: I think, there is no general rule here

File: src/layouts/MenuLayout/MenuLayout.js (hunk 1 above)
Comment (stcherenkov) [reply src/layouts/MenuLayout/MenuLayout.js]: Yeah, but React passes arguments to functions and works with their output, so we just lie to ourselves without reason

Comment (stcherenkov) [reply src/layouts/MenuLayout/MenuLayout.js]: But personally I don't care for those types because we don't use them as intended.

Reverted

File: src/components/MenuPanel/MainAction/MainAction.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (IViol) [reply src/components/MenuPanel/MainAction/MainAction.js]: oh... imagine component with 3 required props, would you like to render separate node for 6 possible combinations?

File: src/layouts/MenuLayout/MenuLayout.js (hunk 1 above)
Comment (IViol) [reply src/layouts/MenuLayout/MenuLayout.js]: > Yeah, but React passes arguments to functions and works with their output

where did you get such info?

> so we just lie to ourselves without reason

we use and call our functions as per we said about it to `flow`. How we lie to ourselves?

File: src/components/MenuPanel/MainAction/MainAction.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (stcherenkov) [reply src/components/MenuPanel/MainAction/MainAction.js]: Highly depends on props
